### PR TITLE
Implementation of Partial Mutation Pitests in pom.xml 

### LINKED ISSUE #19: Implement Partial Pitests
Using PR located at this Link: https://github.com/ucsb-cs156-f24/STARTER-team03/pull/14

Implement Partial Tests for pitest mutations.

### PR DESCRIPTION
**Pull request by ElijahCanderson**
Closes #19 

Implemented the ability to do partial Pitests with the commands provided below. Only some simple edits to pom.xml and edits to readme for the main commands. 

For tests on one class do:
mvn pitest:mutationCoverage -DtargetClasses=edu.ucsb.cs156.organic.controllers.CLASS_NAME

You must use the class name, not the file name.

For tests on a package:

mvn pitest:mutationCoverage -DtargetClasses=edu.ucsb.cs156.organic.controllers.*

Output after running --- : mvn pitest:mutationCoverage -DtargetClasses=edu.ucsb.cs156.organic.controllers.StudentsController

image

Output after running ---:mvn pitest:mutationCoverage -DtargetClasses=edu.ucsb.cs156.organic.controllers.*
image

### CODE REVIEW

**Review by Division7 — DISMISSED** [review]
LGTM 😎 🤙

**Review by pconrad — CHANGES_REQUESTED** [review]
Nice... but please copy over the changes to the README.md as well from the source PR.

**Review by Division7 — DISMISSED** [review]
LGTM! 😎🤙

File: README.md [hunk 1]
```diff
@@ -9,6 +9,27 @@
...
+For example, to run pitest on just one class, use:
+
+```
+mvn pitest:mutationCoverage -DtargetClasses=edu.ucsb.cs156.example.controllers.RestaurantsController
+```
+
+To run pitest on just one package, use:
+
+```
+mvn pitest:mutationCoverage -DtargetClasses=edu.ucsb.cs156.example.controllers.*
```
Comment (pconrad) [inline README.md]:
```suggestion
mvn pitest:mutationCoverage -DtargetClasses=edu.ucsb.cs156.example.controllers.\*
```

The backslash is important.

Also, please change the examples to ones that actually work on the organic codebase; these are just copy/pasted from STARTER-team03.

Comment (ElijahCanderson) [reply README.md]: Appologies, you said copy over so I thought you meant literally copy over.

**Review by GuBinhti — APPROVED** [review]
LGTM

**Pull request merged**